pkg/remoteconfig/orchestrator_k8s_crd.go

```diff
@@ -0,0 +1,126 @@
+package remoteconfig
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+const (
+	crdRcProduct = "ORCHESTRATOR_K8S_CRD"
+)
+
+// CustomResourceDefinitionURLs defines model for CustomResourceDefinitionURLs.
+type CustomResourceDefinitionURLs struct {
+	Crds *[]string `json:"crds,omitempty"`
+}
+
+func (r *RemoteConfigUpdater) crdConfigUpdateCallback(updates map[string]state.RawConfig, applyStatus func(string, state.ApplyStatus)) {
+	ctx := context.Background()
+
+	var configIDs []string
+	for id := range updates {
+		applyStatus(id, state.ApplyStatus{State: state.ApplyStateUnacknowledged, Error: ""})
+		configIDs = append(configIDs, id)
+	}
+
+	mergedUpdate, err := r.parseCRDReceivedUpdates(updates, applyStatus)
+	if err != nil {
+		r.logger.Error(err, "Failed to merge updates")
+		return
+	}
+
+	dda, err := r.getDatadogAgentInstance(ctx)
+	if err != nil {
+		r.logger.Error(err, "Failed to get updatable agents")
+		return
+	}
+
+	if err := r.crdUpdateInstanceStatus(dda, mergedUpdate); err != nil {
+		r.logger.Error(err, "Failed to update status")
+		applyStatus(configIDs[len(configIDs)-1], state.ApplyStatus{State: state.ApplyStateError, Error: err.Error()})
```

Comment:
This looks like it will only update the ApplyState of the last config that was sent. So if three configs get sent, the first two would be "Unacknowledged" and the last one would be "Error". Not sure if this is intended or not. 

For more info on this feature, you can the ApplyState of configs in the Clients page of the [RC Admin app](https://remote-config-admin.us1.prod.dog/?orgId=2&product=AGENT_CONFIG). 
![image](https://github.com/user-attachments/assets/0da59061-bed0-4ade-a14d-dc20e57b233c)

Reply:
Hmm I have to double check this. I was just following the existing pattern here.  Above it does apply status to each config. I need to do some testing to see